pkg/utils/k8sutil.go

```diff
@@ -565,7 +565,8 @@ func UpdateK8sCustomResource(f *spec.Function) error {
 	}
 
 	// TODO: looking for a way to not writing to temp file
-	err = ioutil.WriteFile(".func.json", funcJSON, 0644)
+	filename := os.TempDir() + "/.func.json"
```

Comment:
I don't think it's important that the file end in `.json`(?) - so just create a tempfile directly here, rather than creating a tempdir + file:
```
f, err := ioutil.TempFile("", "kubeless")
if err != nil {
  return err
}
defer os.Remove(f.Name())
if err := f.Write(funcJSON); err != nil {
  return err
}
if err := f.Close(); err != nil {
  return err
}
filename := f.Name()
...
```

Reply:
`os.TempDir()` returns `/tmp` (in linux), it doesn't create a new folder:  https://golang.org/pkg/os/#TempDir